BootloaderCorePkg/Library/S3SaveRestoreLib/S3SaveRestore.c

```diff
@@ -16,8 +16,13 @@
 #include <Guid/SmmInformationGuid.h>
 #include <Base.h>
 
+#if __GNUC__

```

Comment:
The mRegRead/mRegWrite array have function pointers which have different return types. I think this needs to be avoided. Can you refactor this to use simple static function like ReadReg() and WriteReg()?

Reply:
Sure, will be updated.